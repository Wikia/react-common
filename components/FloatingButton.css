/**
* Implemenation taken from http://www.antimath.info/css/sass-sqrt-function/
*/
/**
* Implementation taken from http://alienryderflex.com/hsp.html
*/
/**
* Implemenation taken from http://www.antimath.info/css/sass-sqrt-function/
*/
/**
* Implementation taken from http://alienryderflex.com/hsp.html
*/
/**
* Implemenation taken from http://www.antimath.info/css/sass-sqrt-function/
*/
/**
* Implementation taken from http://alienryderflex.com/hsp.html
*/
.wds-floating-button {
  align-items: center;
  background: #fff;
  border-radius: 50%;
  border: 0;
  box-shadow: 0 0 10px 0 rgba(26, 26, 26, 0.2);
  display: flex;
  height: 36px;
  justify-content: center;
  margin: 0;
  outline: none;
  padding: 0;
  transition-duration: 300ms;
  transition-property: box-shadow;
  width: 36px;
}
.wds-floating-button:not([disabled]), .wds-floating-button:not(.wds-is-disabled) {
  cursor: pointer;
}
.wds-floating-button:not([disabled]):hover, .wds-floating-button:not(.wds-is-disabled):hover {
  box-shadow: 0 0 10px 0 rgba(26, 26, 26, 0.4);
}

.wds-floating-button-group {
  display: inline-flex;
}
.wds-floating-button-group.wds-is-vertical {
  flex-flow: column;
}
.wds-floating-button-group:not(.wds-is-vertical) > .wds-floating-button:not(:first-child) {
  margin-left: 8px;
}
.wds-floating-button-group.wds-is-vertical > .wds-floating-button:not(:first-child) {
  margin-top: 8px;
}