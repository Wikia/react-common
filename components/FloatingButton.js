'use strict';

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var React = _interopDefault(require('react'));
var PropTypes = _interopDefault(require('prop-types'));

function _extends() {
  _extends = Object.assign || function (target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];

      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }

    return target;
  };

  return _extends.apply(this, arguments);
}

function _objectWithoutPropertiesLoose(source, excluded) {
  if (source == null) return {};
  var target = {};
  var sourceKeys = Object.keys(source);
  var key, i;

  for (i = 0; i < sourceKeys.length; i++) {
    key = sourceKeys[i];
    if (excluded.indexOf(key) >= 0) continue;
    target[key] = source[key];
  }

  return target;
}

function _objectWithoutProperties(source, excluded) {
  if (source == null) return {};

  var target = _objectWithoutPropertiesLoose(source, excluded);

  var key, i;

  if (Object.getOwnPropertySymbols) {
    var sourceSymbolKeys = Object.getOwnPropertySymbols(source);

    for (i = 0; i < sourceSymbolKeys.length; i++) {
      key = sourceSymbolKeys[i];
      if (excluded.indexOf(key) >= 0) continue;
      if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
      target[key] = source[key];
    }
  }

  return target;
}

/**
 * Floating button (icons-only)
 */

var FloatingButton = function FloatingButton(_ref) {
  var className = _ref.className,
      href = _ref.href,
      children = _ref.children,
      rest = _objectWithoutProperties(_ref, ["className", "href", "children"]);

  var classes = ['wds-floating-button', className].filter(function (c) {
    return c;
  }).join(' ');

  if (href) {
    return React.createElement("a", _extends({
      href: href,
      className: classes
    }, rest), children);
  }

  return React.createElement("button", _extends({
    className: classes
  }, rest), children);
};

FloatingButton.propTypes = {
  /**
  * @ignore
  */
  children: PropTypes.node,

  /**
  * href attribute.
  * FloatingButton uses `<a>` tag if it's present.
  */
  className: PropTypes.string,

  /**
  * Additional class name
  */
  disabled: PropTypes.bool,

  /**
  * Disabled attribute for the `<button>`
  */
  href: PropTypes.string,

  /**
  * Callback for the `<button>`
  */
  onClick: PropTypes.func
};
FloatingButton.defaultProps = {
  children: null,
  className: '',
  disabled: false,
  href: null,
  onClick: function onClick() {}
};

module.exports = FloatingButton;
