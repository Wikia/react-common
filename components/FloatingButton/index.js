import React from 'react';
import PropTypes from 'prop-types';

import './styles.scss';

/**
 * Floating button (icons-only)
 */
const FloatingButton = ({
    className,
    href,
    children,
    ...rest
}) => {
    const classes = [
        'wds-floating-button',
        className,
    ].filter(c => c).join(' ');

    if (href) {
        return <a href={href} className={classes} {...rest}>{children}</a>;
    }

    return <button className={classes} {...rest}>{children}</button>;
};

FloatingButton.propTypes = {
    /** @ignore */
    children: PropTypes.node,
    /** Additional class name */
    className: PropTypes.string,
    /** Disabled attribute for the `<button>` */
    disabled: PropTypes.bool,
    /** href attribute - floatingButton uses `<a>` tag if it's present */
    href: PropTypes.string,
    /** Callback for the `<button>` */
    onClick: PropTypes.func,
};

FloatingButton.defaultProps = {
    children: null,
    className: '',
    disabled: false,
    href: null,
    onClick: () => {},
};

export default FloatingButton;
