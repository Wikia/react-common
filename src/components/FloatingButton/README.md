Defaults:
```js
<FloatingButton onClick={() => {alert('Clicked!')}}>AZ</FloatingButton>
```
