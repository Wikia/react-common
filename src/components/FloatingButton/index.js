import React from 'react';
import PropTypes from 'prop-types';

/**
 * Floating button (icons-only)
 */
const FloatingButton = ({
  className,
  href,
  children,
  ...rest
}) => {
  const classes = [
    'wds-floating-button',
    className,
  ].filter(c => c).join(' ');

  if (href) {
    return <a href={href} className={classes} {...rest}>{children}</a>;
  }

  return <button className={classes} {...rest}>{children}</button>;
};

FloatingButton.propTypes = {
  /**
   * @ignore
   */
  children: PropTypes.node,
  /**
   * href attribute.
   * FloatingButton uses `<a>` tag if it's present.
   */
  href: PropTypes.string,
  /**
   * Additional class name
   */
  className: PropTypes.string,
  /**
   * Disabled attribute for the `<button>`
   */
  disabled: PropTypes.bool,
  /**
   * Callback for the `<button>`
   */
  onClick: PropTypes.func,
};

FloatingButton.defaultProps = {
  children: null,
  className: '',
  disabled: false,
  href: null,
  onClick: () => {},
};

export default FloatingButton;
