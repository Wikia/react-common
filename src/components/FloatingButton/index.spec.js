import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import sinon from 'sinon';

import FloatingButton from './index';

/* eslint-disable no-alert */

test('FloatingButton renders correctly with default values', () => {
  const component = renderer.create(
    <FloatingButton />,
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('FloatingButton renders correctly with a label', () => {
  const component = renderer.create(
    <FloatingButton>AZ</FloatingButton>,
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('FloatingButton renders correctly as link', () => {
  const component = renderer.create(
    <FloatingButton href="some link">AZ</FloatingButton>,
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('FloatingButton renders correctly with a children', () => {
  /* eslint-disable max-len */
  const component = renderer.create(
    <FloatingButton>
      <svg id="wds-icons-trash-small" viewBox="0 0 18 18" width="100%" height="100%">
        <g fillRule="evenodd">
          <path d="M14 15H4V6h10v9zM6.875 3h4v1h-4V3zM16 4h-3.125V2a1 1 0 0 0-1-1h-6a1 1 0 0 0-1 1v2H2a1 1 0 0 0 0 2v10a1 1 0 0 0 1 1h12a1 1 0 0 0 1-1V6a1 1 0 0 0 0-2z" />
          <path d="M7 14a1 1 0 0 0 1-1V9a1 1 0 0 0-2 0v4a1 1 0 0 0 1 1m4 0a1 1 0 0 0 1-1V9a1 1 0 0 0-2 0v4a1 1 0 0 0 1 1" />
        </g>
      </svg>
    </FloatingButton>,
  );
  /* eslint-enable max-len */
  expect(component.toJSON()).toMatchSnapshot();
});

test('FloatingButton onClick is cliackable', () => {
  const wrapper = shallow(
    <FloatingButton>A</FloatingButton>
  );

  wrapper.find('button').simulate('click');

  expect(true).toBe(true);
});

test('FloatingButton triggers onClick event', () => {
  const mockOnClick = sinon.spy();
  const wrapper = shallow(
    <FloatingButton onClick={mockOnClick}>A</FloatingButton>
  );

  wrapper.find('button').simulate('click');

  expect(mockOnClick.calledOnce).toBe(true);
});
