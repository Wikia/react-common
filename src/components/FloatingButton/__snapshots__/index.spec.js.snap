// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FloatingButton renders correctly as link 1`] = `
<a
  className="wds-floating-button"
  disabled={false}
  href="some link"
  onClick={[Function]}
>
  AZ
</a>
`;

exports[`FloatingButton renders correctly with a children 1`] = `
<button
  className="wds-floating-button"
  disabled={false}
  onClick={[Function]}
>
  <svg
    height="100%"
    id="wds-icons-trash-small"
    viewBox="0 0 18 18"
    width="100%"
  >
    <g
      fillRule="evenodd"
    >
      <path
        d="M14 15H4V6h10v9zM6.875 3h4v1h-4V3zM16 4h-3.125V2a1 1 0 0 0-1-1h-6a1 1 0 0 0-1 1v2H2a1 1 0 0 0 0 2v10a1 1 0 0 0 1 1h12a1 1 0 0 0 1-1V6a1 1 0 0 0 0-2z"
      />
      <path
        d="M7 14a1 1 0 0 0 1-1V9a1 1 0 0 0-2 0v4a1 1 0 0 0 1 1m4 0a1 1 0 0 0 1-1V9a1 1 0 0 0-2 0v4a1 1 0 0 0 1 1"
      />
    </g>
  </svg>
</button>
`;

exports[`FloatingButton renders correctly with a label 1`] = `
<button
  className="wds-floating-button"
  disabled={false}
  onClick={[Function]}
>
  AZ
</button>
`;

exports[`FloatingButton renders correctly with default values 1`] = `
<button
  className="wds-floating-button"
  disabled={false}
  onClick={[Function]}
/>
`;
